packages/devtools_app/lib/src/screens/memory/panes/leaks/instrumentation/model.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+
+enum LeakType {
+  notDisposed,
+  notGCed,
+  gcedLate,
+}
+
+LeakType _parseLeakType(String source) =>
+    LeakType.values.firstWhere((e) => e.toString() == source);
+
+/// Statistical information about found leaks.
+class LeakSummary {
+  LeakSummary(this.totals);
+
+  factory LeakSummary.fromJson(Map<String, dynamic> json) => LeakSummary(
+        json.map(
+          (key, value) => MapEntry(_parseLeakType(key), int.parse(value)),
+        ),
+      );
+
+  final Map<LeakType, int> totals;
+
+  bool get isEmpty => totals.values.sum == 0;
+
+  String toMessage() {
+    return '${totals.values.sum} memory leak(s): '
+        'not disposed: ${totals[LeakType.notDisposed]}, '
+        'not GCed: ${totals[LeakType.notGCed]}, '
+        'GCed late: ${totals[LeakType.gcedLate]}';
+  }
+
+  bool equals(LeakSummary? other) {
```

Comment:
override == instead of creating a method `equals`

Reply:
I prefer 'equals', because the reader of the code that uses '==' will not know if '==' is overriden or not.

It would make sense to override '==' if we used the objects as map keys or something like this.

Reply:
We override `==` in many places in DevTools, so it would be completely reasonable to do this here, too. If you want `==` to continue to compare object identity, then let's rename this method to `matches`. It is confusing to have `equals` and `==` that do not do the same thing. Better to have the term "equals" mean true equality as determined by `==` than to have two different meanings.

Reply:
After some thinking I realized there is no value in comparing LeakSummary by identity. Thanks.

Reply:
Then I've read the code if (oldSummary == newSummary) and realized that to verify this code I need to make sure '==' is overriden. So, created the method 'matches'.